docs/developer/example-bpf.md

```diff
@@ -0,0 +1,670 @@
+`bpfctl` also sends gRPC requests over mTLS to perform actions and request data from `bpfd`.
+4. Once the `go-xdp-counter` BPF bytecode is loaded, the BPF program will write packet counts
+and byte counts to a shared map.
+5. `go-xdp-counter` userspace program periodically reads counters from the shared map and logs
+the value.
+
+#### Running Privileged
+
+The most basic example, just use `sudo` to start the `go-xdp-counter` program.
+Determine the host interface to attach the BPF program to and then start the go program with:
+
+```console
+    cd src/bpfd/examples/go-xdp-counter/
+    sudo ./go-xdp-counter -iface <INTERNET INTERFACE NAME>
+```
+or (**NOTE:** TC programs also require a direction, ingress or egress)
+```console
+    cd src/bpfd/examples/go-tc-counter/
+    sudo ./go-tc-counter -direction ingress -iface <INTERNET INTERFACE NAME>
+```
+
+The output should show the count and total bytes of packets as they pass through the
+interface as shown below:
+
+```console
+    sudo ./go-xdp-counter -iface ens3
+    2022/12/02 15:59:34 Reading /etc/bpfd/gocounter.toml ...
+    2022/12/02 15:59:34 Read /etc/bpfd/gocounter.toml failed: err open /etc/bpfd/gocounter.toml: no such file or directory
+    2022/12/02 15:59:34 Using Input: Interface=ens3 Priority=50 Source=bpf_bpfel.o
+    2022/12/02 15:59:35 Program registered with b6b2107c-f1a3-48ac-a145-1073c0979ba4 id
+    2022/12/02 15:59:38 0 packets received
+    2022/12/02 15:59:38 0 bytes received
+
+    2022/12/02 15:59:41 4 packets received
+    2022/12/02 15:59:41 580 bytes received
+
+    :
+```
+
+Use `bpfctl` to show the `go-xdp-counter` BPF bytecode was loaded.
+
+```console
+    bpfctl list
+    UUID                                  Type  Name   Path                                                      Metadata
+    b6b2107c-f1a3-48ac-a145-1073c0979ba4  xdp   stats  /home/$USER/src/bpfd/examples/go-xdp-counter/bpf_bpfel.o  { "priority": 50, "iface": "ens3", "position": 0, "proceed_on": ["pass", "dispatcher_return"] }
+```
+
+Finally, press `<CTRL>+c` when finished with `go-xdp-counter`.
+
+```console
+    :
+
+    2022/12/02 16:00:56 64 packets received
+    2022/12/02 16:00:56 9280 bytes received
+
+    2022/12/02 16:00:59 64 packets received
+    2022/12/02 16:00:59 9280 bytes received
+
+    ^C2022/12/02 16:01:00 Exiting...
+    2022/12/02 16:01:00 Unloading Program: b6b2107c-f1a3-48ac-a145-1073c0979ba4
+```
+
+#### Running Unprivileged
+
+To run the `go-xdp-counter` and `go-tc-counter` examples unprivileged (without `sudo`), the
+following three steps must be performed.
+
+##### Step 1: Create `bpfd` User Group
+
+The [tutorial.md](../admin/tutorial.md) guide describes the different modes `bpfd` can be run in.
+Specifically, the [Systemd Service](../admin/tutorial.md#systemd-service) section describe how to
+start `bpfd` the `bpfd` Users and `bpfd` User Group.
+`bpfd` must be started as a Systemd Service and `go-xdp-counter` must be run from a User that is a
```

Comment:
All this about the shared Socket is stale right? We got rid of passing the map FD over a socket

Reply:
I didn't realize it was gone. Removed documentation on it and also the const in both main.c files